toolkit/scripts/check_required_kernel_configs.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for required kernel configs in a kernel config file
+# Usage: python3 check_required_kernel_configs.py --required_configs <path to json of required configs> --config_file <path to config being checked>
+
+import json
+import argparse
+import sys
+import re
+
+# Define a class
+    #incorrect configs is map: {config: (newValue, expectedValue, comment, PR)}
+class IncorrectConfig:
+    def __init__(self, name, newValue, expectedValue, comment, PR):
+        self.name = name
+        self.newValue = newValue
+        self.expectedValue = expectedValue
+        self.comment = comment
+        self.PR = PR
+
+
+def get_data_from_config(input_file):
+    with open(input_file, 'r') as file:
+        input_config_data = file.read()
+    return input_config_data
+
+def check_kernel(input_file):
+    match = re.search(r'SPECS/(.*?)/', input_file)
+    if match:
+        return match.group(1)
+    else:
+        return None
```

Comment:
It looks like you have some duplicated functions here and in the other script. I'd suggest placing them in a shared module, so we don't repeat ourselves. This also makes it harder to update one but forget the other.

Reply:
Discussion notes: low-pri, might want to consider to consolidate both scripts into one "kernel config checker".